evalml/tests/pipeline_tests/test_pipeline_utils.py

```diff
@@ -150,6 +151,18 @@ def test_make_pipeline(
             email_featurizer = [EmailFeaturizer] if "email" in column_names else []
             url_featurizer = [URLFeaturizer] if "url" in column_names else []
             imputer = [] if (column_names in [["ip"], ["all_null"]]) else [Imputer]
+            drop_nan_rows_transformer = (
+                [DropNaNRowsTransformer]
+                if is_time_series(problem_type)
+                and estimator_class.model_family
+                in [
+                    ModelFamily.EXTRA_TREES,
+                    ModelFamily.RANDOM_FOREST,
+                    ModelFamily.LINEAR_MODEL,
+                    ModelFamily.DECISION_TREE,
+                ]
```

Comment:
I think we should strongly consider what I mentioned before, about making this a function, so that we don't accidentally diverge in who needs this transformer.